### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Update DefaultNewLine.  It's now a private const.
- Value changed to \n, to follow .NET implementation see [here](https://docs.microsoft.com/en-us/dotnet/api/system.io.ports.serialport.newline?view=dotnet-plat-ext-5.0#System_IO_Ports_SerialPort_NewLine).
- Replace local InfiniteTimeout with Threading.Timeout.Infinite.
- Remove Encoding property
  + No point on exposing it, as nanoFramework only supports UTF8 encoding.
  + Also allows removing dependency of System.Text.
- Add missing NewLine backing field.
- Add using for System.Threading (smplified access to object within).
- Implement getter for WatchChar.
- Rework native calls:
   + Replace native calls with more obvious one for the "new" IO.Ports API.
   + Update SerialStream accordingly.
- Add pragmas to keep sonarcloud analyser happy.
- Bump AssemblyNativeVersion to 100.1.2.0.

## Motivation and Context
- Align API with full .NET.
- Performance improvement by tossing the code of the various handlers to native code.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: System.IO.Ports/SerialPort.cs [hunk 1]
```diff
@@ -14,31 +13,25 @@ namespace System.IO.Ports
...
-        /// </summary>
-        public const int InfiniteTimeout = -1;
-
-        /// <summary>
-        /// Default new Line, you can change and adjust it in the NewLine property
-        /// </summary>
-        public const string DefaultNewLine = "\r\n";
+        // default new line
+        [System.Diagnostics.DebuggerBrowsable(Diagnostics.DebuggerBrowsableState.Never)]
+        private const string _defaultNewLine = "\r";
```
Comment (Ellerbach) [inline System.IO.Ports/SerialPort.cs:18]: Why removing the default const one?

File: System.IO.Ports/packages.config [hunk 2]
```diff
@@ -2,6 +2,5 @@
 <packages>
   <package id="nanoFramework.CoreLibrary" version="1.10.5" targetFramework="netnanoframework10" />
   <package id="nanoFramework.Runtime.Events" version="1.9.1" targetFramework="netnanoframework10" />
-  <package id="nanoFramework.System.Text" version="1.1.1" targetFramework="netnanoframework10" />
```
Comment (Ellerbach) [inline System.IO.Ports/packages.config:5]: You should adjust the nuspec as well

**Review by Ellerbach — COMMENTED** [review]
All up looks good, please don't forget to adjust the nuspec ;-)

File: System.IO.Ports/SerialPort.cs (hunk 1 above)
Comment (josesimoes) [reply System.IO.Ports/SerialPort.cs:18]: Not removing. Just adjusting to follow .NET (see the PR comment, second bullet).

**Pull request merged**